lib/devicetrust/authn/authn.go

```diff
@@ -26,23 +26,37 @@ import (
 	"github.com/gravitational/teleport/lib/devicetrust/native"
 )
 
-// vars below are used to swap native methods for fakes in tests.
-var (
-	getDeviceCredential          = native.GetDeviceCredential
-	collectDeviceData            = native.CollectDeviceData
-	signChallenge                = native.SignChallenge
-	solveTPMAuthnDeviceChallenge = native.SolveTPMAuthnDeviceChallenge
-	getDeviceOSType              = native.GetDeviceOSType
-)
+// Ceremony is the device authentication ceremony.
+// It takes the client role of
+// [devicepb.DeviceTrustServiceClient.AuthenticateDevice]
+type Ceremony struct {
+	GetDeviceCredential          func() (*devicepb.DeviceCredential, error)
+	CollectDeviceData            func() (*devicepb.DeviceCollectedData, error)
+	SignChallenge                func(chal []byte) (sig []byte, err error)
+	SolveTPMAuthnDeviceChallenge func(challenge *devicepb.TPMAuthenticateDeviceChallenge) (*devicepb.TPMAuthenticateDeviceChallengeResponse, error)
+	GetDeviceOSType              func() devicepb.OSType
+}
+
+// NewCeremony creates a new ceremony that delegates per-device behavior
+// to lib/devicetrust/native.
+// If you want to customize a [Ceremony], for example for testing purposes, you
+// may create a configure an instance directly, without calling this method.
+func NewCeremony() *Ceremony {
+	return &Ceremony{
+		GetDeviceCredential: native.GetDeviceCredential,
+		CollectDeviceData:   native.CollectDeviceData,
+		SignChallenge:       native.SignChallenge,
```

Comment:
It doesn't look like we are initializing all fields here.

```suggestion
		GetDeviceCredential:          native.GetDeviceCredential,
		CollectDeviceData:            native.CollectDeviceData,
		SignChallenge:                native.SignChallenge,
		SolveTPMAuthnDeviceChallenge: native.SolveTPMAuthnDeviceChallenge,
		GetDeviceOSType:              native.GetDeviceOSType,
```